frontend/test/metabase/lib/expressions/typechecker.unit.spec.js

```diff
@@ -142,4 +143,55 @@ describe("type-checker", () => {
       expect(aggregation("[X]+Sum([Y])").metrics).toEqual(["X"]);
     });
   });
+
+  describe("compactSyntaxTree", () => {
```

Comment:
Note: it's not wired up to the type checker yet, so the unit tests are the best way to look at how it could be useful.

Reply:
The tests look good!